include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -28,406 +62,665 @@
+    }
 
+    inline void unserialize(value_type& scalar, const dataspace_type* data) {
+        scalar = Reference(*data);
+    }
 
-// apply conversion for vectors 1D
-template <typename T>
-struct data_converter<
-    std::vector<T>,
-    typename std::enable_if<(
-        std::is_same<T, typename type_of_array<T>::type>::value &&
-        !std::is_same<T, Reference>::value
-        )>::type>
-    : public container_converter<std::vector<T>> {
-
-    using container_converter<std::vector<T>>::container_converter;
-};
+    inline void serialize(const value_type& scalar, dataspace_type* data) {
+        scalar.create_ref(data);
+    }
+
+    const DataSpace& _space;
+    size_t _number_of_element = 1;
 
+    static constexpr size_t number_of_dims = 0;
+};
 
-// apply conversion to std::array
-template <typename T, std::size_t S>
-struct data_converter<
-    std::array<T, S>,
-    typename std::enable_if<(
-        std::is_same<T, typename type_of_array<T>::type>::value)>::type>
-    : public container_converter<std::array<T, S>> {
+template<>
+struct data_converter<std::string> {
+    using value_type = std::string;
+    using dataspace_type = const char*;
+    using h5_type = value_type;
 
-    inline data_converter(const DataSpace& space)
-        : container_converter<std::array<T, S>>(space)
+    inline data_converter(const DataSpace& space, const std::vector<size_t>& dims)
+      : _space(space)
     {
-        auto&& dims = space.getDimensions();
-        if (!is_1D(dims)) {
-            throw DataSpaceException("Only 1D std::array supported currently.");
-        }
-        if (compute_total_size(dims) != S) {
-            std::ostringstream ss;
-            ss << "Impossible to pair DataSet with " << compute_total_size(dims)
-               << " elements into an array with " << S << " elements.";
-            throw DataSpaceException(ss.str());
+        if (!dims.empty()) {
+            throw std::string("Invalid number of elements for variable string");
         }
     }
 
-    inline T* transform_read(std::array<T, S>& vec) const noexcept {
-        return vec.data();
+    void allocate(value_type&) {
+        // pass
+    }
+
+    static std::vector<size_t> get_size(const value_type& val) {
+        return std::vector<size_t>{};
+    }
+
+    static dataspace_type* get_pointer(value_type& val) {
+        // .data() return a const pointer until c++17
+        throw std::string("Invalid get_pointer on std::string");
     }
+
+    static const dataspace_type* get_pointer(const value_type& val) {
+        static dataspace_type _c_str = const_cast<char*>(val.c_str());
+        return &_c_str;
+    }
+
+    inline void unserialize(value_type& scalar, const dataspace_type* data) {
+        scalar = std::string(*data);
+    }
+
+    inline void serialize(const value_type& scalar, dataspace_type* data) {
+        dataspace_type _c_str = const_cast<char*>(scalar.c_str());
+        *data = _c_str;
+    }
+
+    const DataSpace& _space;
```

Comment:
Generic comment/question: I thought that `_xyz` was frowned upon in C communities? Shouldn't the `_` come at the end?

Reply:
only if this capital or _ after the first _.

Reply:
I have no thought on it. Help me with this class!